### ISSUE TITLE
How to fix TIMIT_TrainAutoEncoder : same value per frame in output file

### ISSUE THREAD

**Issue by Kaya122** [Microsoft/CNTK#472]
Dear all.

I have tried to train the denoised auto encoder with TIMIT_TrainAutoEncoder 
There is no error show in the log while it is training and writing,
but the result of the denoised auto encoder is strange because of the same value per frame in output file.

So I am sure that the training procedure has something wrong.
So I would like to ask you how to fix of it
I added my configure to here, but the contenst is so long 
so I just add main things..

First I trained TIMIT_TrainAutoEncoder 

 1 stderr=$ExpDir$/log
  2 command=TIMIT_TrainAutoEncoder
  3
  4 precision=float
  5
  6 #######################################
  7 #  TRAINING CONFIG (Simple, Fixed LR) #
  8 #######################################
  9
 10 TIMIT_TrainAutoEncoder=[
 11     action=train
 12
 13     modelPath=$ExpDir$/cntkSpeech.dnn
 14
 15     # deviceId=-1 for CPU, >=0 for GPU devices
 16     deviceId=$DeviceNumber$
 17
 18     traceLevel=1
 19
 20     # notation xxx:yyy_n:zzz is equivalent to xxx, then yyy repeated n times, then z    zz
 21     # example: 10:20_3:5 is equivalent to 10:20:20:20:5
 22     NDLNetworkBuilder=[
 23         ndlMacros=$NdlDir$/default_macros.ndl
 24         NetworkDescription=$NdlDir$/ae.ndl
 25     ]
 26
 27     SGD=[
 28         # epochSize=0 means epochSize is the size of the training set
 29         epochSize=0
 30         minibatchSize=256
 31         learningRatesPerMB=0.1
 32         momentumPerMB=0.9
 33         dropoutRate=0.0
 34         maxEpochs=25
 35     ]
 36
 37     # Parameter values for the reader
 38     reader=[
 39       # reader to use
 40       readerType=HTKMLFReader
 41
 42           readMethod=blockRandomize
 43           randomize=500000000
 44 #      readMethod=rollingWindow
 45
 46       miniBatchMode=Partial
 47       randomize=Auto
 48       verbosity=1
 49
 50       featIn=[
 51                 dim=280
 52                 scpFile=$ScpDir$/mfc_clean_SMCT1_40.ach
 53       ]
 54
 55       featOut=[
 56                 scpFile=$ScpDir$/mfc_clean_40.ach
 57                 dim=40
 58       ]
 59     ]
 60 ]

and the log file is like that 
 ##############################################################################
   181 #                                                                            #
   182 # Action "train"                                                             #
   183 #                                                                            #
   184 ##############################################################################
   185
   186 CNTKCommandTrainBegin: TIMIT_TrainAutoEncoder
   187 NDLBuilder Using GPU 0
   188 reading script file /home/htk/DNN_DAE/DAE_LMFE_SPLICE_3_batch_mel2mel_SMCT_0dB_tr       y_train/scp/mfc_clean_SMCT1_40.ach ... 7137 entries
   189 reading script file /home/htk/DNN_DAE/DAE_LMFE_SPLICE_3_batch_mel2mel_SMCT_0dB_tr       y_train/scp/mfc_clean_40.ach ... 7137 entries
   190 .................................................................................       ...................feature set 0: 5438018 frames in 7137 out of 7137 utterances
   191 minibatchutterancesource: 7137 utterances grouped into 61 chunks, av. chunk size:        117.0 utterances, 89147.8 frames
   192 .................................................................................       ...................feature set 1: 5438018 frames in 7137 out of 7137 utterances
   193 minibatchutterancesource: 7137 utterances grouped into 61 chunks, av. chunk size:        117.0 utterances, 89147.8 frames
   194
   195 Creating virgin network.
   196 SetUniformRandomValue (GPU): creating curand object with seed 1, sizeof(ElemType)       ==4
   197
   198 Post-processing network...
   199
   200 8 roots:
   201         MeanSqErr.BFF.FF.P = Plus()
   202         MeanSqErr.MSE = SquareError()
   203         featNormIn.xMean = Mean()
   204         featNormIn.xNorm = PerDimMeanVarNormalization()
   205         featNormIn.xStdDev = InvStdDev()
   206         featNormOut.xMean = Mean()
   207         featNormOut.xNorm = PerDimMeanVarNormalization()
   208         featNormOut.xStdDev = InvStdDev()
   209
   210 Validating network. 33 nodes to process in pass 1.
   211
   212 Validating --> MeanSqErr.BFF.W = LearnableParameter() :  -> [40 x 1024]
   213 Validating --> L4.BFF.W = LearnableParameter() :  -> [1024 x 1024]
   214 Validating --> L3.BFF.W = LearnableParameter() :  -> [1024 x 1024]
   215 Validating --> L2.BFF.W = LearnableParameter() :  -> [1024 x 1024]
   216 Validating --> L1.BFF.W = LearnableParameter() :  -> [1024 x 280]
   217 Validating --> featIn = InputValue() :  -> [280 x *]
   218 Validating --> L1.BFF.FF.T = Times (L1.BFF.W, featIn) : [1024 x 280], [280 x *] -       > [1024 x *]
   219 Validating --> L1.BFF.B = LearnableParameter() :  -> [1024]
   220 Validating --> L1.BFF.FF.P = Plus (L1.BFF.FF.T, L1.BFF.B) : [1024 x *], [1024] ->        [1024 x *]
   221 Validating --> L1.S = Sigmoid (L1.BFF.FF.P) : [1024 x *] -> [1024 x *]
Validating --> L2.BFF.FF.T = Times (L2.BFF.W, L1.S) : [1024 x 1024], [1024 x *] -       > [1024 x *]
   223 Validating --> L2.BFF.B = LearnableParameter() :  -> [1024]
   224 Validating --> L2.BFF.FF.P = Plus (L2.BFF.FF.T, L2.BFF.B) : [1024 x *], [1024] ->        [1024 x *]
   225 Validating --> L2.S = Sigmoid (L2.BFF.FF.P) : [1024 x *] -> [1024 x *]
   226 Validating --> L3.BFF.FF.T = Times (L3.BFF.W, L2.S) : [1024 x 1024], [1024 x *] -       > [1024 x *]
   227 Validating --> L3.BFF.B = LearnableParameter() :  -> [1024]
   228 Validating --> L3.BFF.FF.P = Plus (L3.BFF.FF.T, L3.BFF.B) : [1024 x *], [1024] ->        [1024 x *]
   229 Validating --> L3.S = Sigmoid (L3.BFF.FF.P) : [1024 x *] -> [1024 x *]
   230 Validating --> L4.BFF.FF.T = Times (L4.BFF.W, L3.S) : [1024 x 1024], [1024 x *] -       > [1024 x *]
   231 Validating --> L4.BFF.B = LearnableParameter() :  -> [1024]
   232 Validating --> L4.BFF.FF.P = Plus (L4.BFF.FF.T, L4.BFF.B) : [1024 x *], [1024] ->        [1024 x *]
   233 Validating --> L4.S = Sigmoid (L4.BFF.FF.P) : [1024 x *] -> [1024 x *]
   234 Validating --> MeanSqErr.BFF.FF.T = Times (MeanSqErr.BFF.W, L4.S) : [40 x 1024],        [1024 x *] -> [40 x *]
   235 Validating --> MeanSqErr.BFF.B = LearnableParameter() :  -> [40]
   236 Validating --> MeanSqErr.BFF.FF.P = Plus (MeanSqErr.BFF.FF.T, MeanSqErr.BFF.B) :        [40 x *], [40] -> [40 x *]
   237 Validating --> featOut = InputValue() :  -> [40 x *]
   238 Validating --> MeanSqErr.MSE = SquareError (featOut, MeanSqErr.BFF.FF.P) : [40 x        *], [40 x *] -> [1]
   239 Validating --> featNormIn.xMean = Mean (featIn) : [280 x *] -> [280]
   240 Validating --> featNormIn.xStdDev = InvStdDev (featIn) : [280 x *] -> [280]
   241 Validating --> featNormIn.xNorm = PerDimMeanVarNormalization (featIn, featNormIn.       xMean, featNormIn.xStdDev) : [280 x *], [280], [280] -> [280 x *]
   242 Validating --> featNormOut.xMean = Mean (featOut) : [40 x *] -> [40]
   243 Validating --> featNormOut.xStdDev = InvStdDev (featOut) : [40 x *] -> [40]
   244 Validating --> featNormOut.xNorm = PerDimMeanVarNormalization (featOut, featNormO       ut.xMean, featNormOut.xStdDev) : [40 x *], [40], [40] -> [40 x *]
   245
   246 Validating network. 21 nodes to process in pass 2.
   247
   248
   249 Validating network, final pass.
1
   252
   253 15 out of 33 nodes do not share the minibatch layout with the input data.
   254
   255 Post-processing network complete.
   256
   257 Created model with 33 nodes on GPU 0.
   258
   259 Training criterion node(s):
   260         MeanSqErr.MSE = SquareError
   261
   262
   263 Allocating matrices for forward and/or backward propagation.
   264
   265 Memory Sharing Structure:
   266
   267 (nil): {[featIn Gradient[280 x *]] [featNormIn.xMean Gradient[280]] [featNormIn.x       Norm Gradient[280 x *]] [featNormIn.xNorm Value[280 x *]] [featNormIn.xStdDev Gra       dient[280]] [featNormOut.xMean Gradient[40]] [featNormOut.xNorm Gradient[40 x *]]        [featNormOut.xNorm Value[40 x *]] [featNormOut.xStdDev Gradient[40]] [featOut Gr       adient[40 x *]] }
   268 0x2f87d80: {[featIn Value[280 x *]] }
   269 0x3153e80: {[L1.BFF.FF.T Gradient[1024 x *]] [L1.S Value[1024 x *]] }
   270 0x34e55b0: {[featOut Value[40 x *]] }
   271 0x39f8c40: {[featNormIn.xMean Value[280]] }
   272 0x39f9130: {[featNormIn.xStdDev Value[280]] }
   273 0x39f9fd0: {[featNormOut.xMean Value[40]] }
   274 0x39fa470: {[featNormOut.xStdDev Value[40]] }
   275 0x39fb1e0: {[L1.BFF.B Value[1024]] }
   276 0x39fbcc0: {[L1.BFF.W Value[1024 x 280]] }

**Comment by cha-zhang** [comment]
I'm closing this issue since it has been too old. @Kaya122 if you still working on this and hasn't find answers, please post a new issue. Thanks!
